jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -346,12 +353,29 @@ public String getDefaultEnvironmentName()
         if (defaultEnvironmentName == null)
         {
             return trackedEnvironments.stream()
-                .min(ENVIRONMENT_COMPARATOR)
+                .max(Comparator.comparing(TrackedEnv::weight))
+                .map(TrackedEnv::name)
                 .orElse(null);
         }
         return defaultEnvironmentName;
     }
 
+    protected List<String> getTrackedEnvironmentsByWeight()
+    {
+        return trackedEnvironments.stream()
+            .sorted(Comparator.comparing(TrackedEnv::weight).reversed())
+            .map(TrackedEnv::name)
+            .toList();
+    }
+
+    protected void addTrackedEnvironment(String name, int weight)
+    {
+        if (hasTrackedEnvironment(name))
+            throw new IllegalStateException("Environment [" + name + "] is already configured for deploy.");
+
+        trackedEnvironments.add(new TrackedEnv(name, weight));
+    }
+
     public void setDefaultEnvironmentName(String name)
```

Comment:
This method should be removed there is no concept of default, if not determined by the environments that are enabled.